fiftyone/core/odm/dataset.py

```diff
@@ -106,9 +106,11 @@ class DatasetDocument(Document):
     evaluations = DictField(
         EmbeddedDocumentField(document_type=RunDocument), default=dict
     )
+    evaluation_results = DictField(default=dict)
     brain_methods = DictField(
         EmbeddedDocumentField(document_type=RunDocument), default=dict
     )
+    brain_results = DictField(default=dict)
```

Comment:
`evaluations` and `evaluation_results` (and `brain_methods` and `brain_results`) are one-to-one, correct? I'm wondering if we can make `results` a field on `RunDocument`.

Reply:
I would then use reference fields on a separate. I don't love the idea of leveraging mongoengine but the API is there, and access does not need to be particularly fast here.

Reply:
Yeah info and results are 1-1, so you're welcome to adopt the convention you mention in `odm/dataset.py` if that's cleaner.

The one note is that `Results` can be quite gigantic, which is why there are separate `RunInfo` and `RunResults` user-facing classes. My thinking was that the user should be able to call `print(samples.get_evaluation_info(eval_key))` to quickly get info about the run, and then call `results = samples. load_evaluation_results(eval_key)` separately to actually get the results if desired.

Only saying this b/c I wouldn't want `samples.get_evaluation_info()` to require loading the results part.